File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode current1 = l1;
        ListNode current2 = l2;
        ListNode dummy = new ListNode();
        ListNode current = dummy;
        int carry = 0;
        while(current1!=null || current2!=null){
            int sum = carry;
            if(current1!=null)  sum+= current1.val;
            if(current2!=null)  sum+= current2.val;
            ListNode nextnode = new ListNode(sum%10);
            current.next = nextnode;
            carry = sum/10;
            if(current1!=null)  current1 = current1.next;
            if(current2!=null)  current2 = current2.next;
            current = current.next;
        }
        if(carry>0){
            ListNode nextnode = new ListNode(carry);
            current.next = nextnode;
        }
        return dummy.next;
    }
}

File: 0002-add-two-numbers/0002-add-two-numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        int carry = 0;
        ListNode* p1 = l1;
        ListNode* p2 = l2;
        ListNode* dummy = new ListNode();
        ListNode* ans= dummy;
        while(p1 != nullptr || p2!=nullptr){
            int sum = carry;
            if(p1) {
                sum+= p1->val;
                p1 = p1->next;
            };
            if(p2){
                sum+= p2->val;
                p2 = p2->next;
            };

            ListNode* newnode = new ListNode(sum%10);
            carry = sum/10;
            ans->next = newnode;
            ans = newnode;

        }
        if(carry){
              ListNode* newnode = new ListNode(carry);
              ans->next = newnode;
        }
        return dummy->next;
    }
};